lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -5206,30 +5128,41 @@ class JVPEmitter final
     */
 
     // Initialize tangent mapping for parameters.
+    auto diffLoc = differential.getLocation();
     auto diffParamsIt = getIndices().parameters->begin();
-    for (auto index : range(diffParamArgs.size())) {
-      auto *diffArg = diffParamArgs[index];
+    for (unsigned i : range(diffParamArgs.size())) {
+      auto *diffArg = diffParamArgs[i];
       auto *origArg = origParamArgs[*diffParamsIt];
       diffParamsIt++;
-      if (diffArg->getType().isAddress()) {
-        setTangentBuffer(origEntry, origArg, diffArg);
-      } else {
-        setTangentValue(
-            origEntry, origArg, makeConcreteTangentValue(diffArg));
-      }
+
+      // Create a local copy of the differential argument so that it can be
+      // written to by later differential emission logic. The argument itself
+      // should not be overwritten.
+      diffBuilder.setInsertionPoint(
+          differential.getEntryBlock(),
+           getNextDifferentialLocalAllocationInsertionPoint());
```

Comment:
```suggestion
          getNextDifferentialLocalAllocationInsertionPoint());
```

Reply:
Fixed in https://github.com/apple/swift/pull/26709/commits/bc2657784cb232a2f85921b8446883762a079f18.